src/extensions.rs

```diff
@@ -270,6 +271,78 @@ impl RatchetTreeExtension {
     }
 }
 
+#[repr(u32)]
+#[cfg_attr(rustfmt, rustfmt_skip)]
+pub enum DeviceCapabilityType {
+    AddCap          = 0b0001u32,
+    RemoveCap       = 0b0010u32,
+    NonRemovableCap = 0b0100u32,
+    SelfUpdateCap   = 0b1000u32,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct DeviceCapabilities(pub u32);
+
+impl DeviceCapabilities {
+    pub fn is_permanent(&self) -> bool {
```

Comment:
All these functions should just call a single function that is `match` on `self.0`.

Reply:
I don't think that's going to be much nicer. The `match` statement would have to return a set of `DeviceTypes` and `DeviceCapabilityTypes`. Then the individual function would have to check if the specific `DeviceType` or `DeviceCapabilityType` is among those returned. Is that what you meant?